components/org.wso2.carbon.identity.application.authenticator.adapter/src/main/java/org/wso2/carbon/identity/application/authenticator/adapter/util/AuthenticatedUserBuilder.java

```diff
@@ -144,29 +198,32 @@ private ClaimMapping buildClaimMapping(String claimUri) {
         return claimMapping;
     }
 
-    private AbstractUserStoreManager resolveUserStoreManager(AuthenticationContext context, String userStoreDomain)
+    private AbstractUserStoreManager resolveUserStoreManager(AuthenticatedUserData.UserStore userStore)
             throws ActionExecutionResponseProcessorException {
 
         AbstractUserStoreManager userStoreManager;
-
         try {
             RealmService realmService = AuthenticatorAdapterDataHolder.getInstance().getRealmService();
             int tenantId = IdentityTenantUtil.getTenantId(context.getTenantDomain());
             UserRealm userRealm = (UserRealm) realmService.getTenantUserRealm(tenantId);
-            if (StringUtils.isNotBlank(userStoreDomain)) {
+            if (userStore != null && StringUtils.isNotBlank(userStore.getName())) {
                 userStoreManager = (AbstractUserStoreManager) userRealm.getUserStoreManager()
-                        .getSecondaryUserStoreManager(userStoreDomain);
+                        .getSecondaryUserStoreManager(userStore.getName());
             } else {
                 userStoreManager = (AbstractUserStoreManager) userRealm.getUserStoreManager();
             }
         } catch (UserStoreException e) {
-            throw new ActionExecutionResponseProcessorException("An error occurs when trying to retrieve the " +
-                    "userStore manager for the given userStore domain name:" +  userStoreDomain, e);
+            if (userStore != null && StringUtils.isNotBlank(userStore.getName())) {
+                throw new ActionExecutionResponseProcessorException(String.format("An error occurred while " +
+                        "retrieving the userStore manager for the given userStore domain: %s.", userStore.getName()),
+                        e);
+            }
+            throw new ActionExecutionResponseProcessorException("An error occurred while retrieving " +
+                    "the userStore manager the given userStore domain.", e);
```

Comment:
There's a slight grammatical problem.
An error occurred while retrieving the userStore manager *for* the given userStore domain.

Reply:
Also doesn't both convey the same message.
I think if you are checking for the empty user store the latter should be failed to retrieve the primary which is the default